WordPress/Classes/Utility/Gravatar.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+struct Gravatar {
+    private struct Defaults {
+        static let scheme = "https"
+        static let host = "secure.gravatar.com"
+        // unknownHash = md5("unknown@gravatar.com")
+        static let unknownHash = "ad516503a11cd5ca435acc9bb6523536"
+    }
+
+    let canonicalURL: NSURL
+
+    func urlWithSize(size: Int) -> NSURL {
+        let components = NSURLComponents(URL: canonicalURL, resolvingAgainstBaseURL: false)!
+        components.query = "s=\(size)&d=404"
+        return components.URL!
+    }
+
+    static func isGravatarURL(url: NSURL) -> Bool {
+        guard let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) else {
+            return false
+        }
+
+        guard let host = components.host
+            where host.hasSuffix(".gravatar.com") else {
+                return false
+        }
+
+        guard let path = url.path
+            where path.hasPrefix("/avatar/") else {
+                return false
+        }
+
+        return true
+    }
+}
+
+extension Gravatar: Equatable {}
+
+func ==(lhs: Gravatar, rhs: Gravatar) -> Bool {
+    return lhs.canonicalURL == rhs.canonicalURL
+}
+
+extension Gravatar {
```

Comment:
From what I understood, that was the right way of building "convenience initializers" for structs. IIRC, if you declare it in the struct, it won't generate the default initializer.

See https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Extensions.html#//apple_ref/doc/uid/TP40014097-CH24-ID153